### PR TITLE
coq: 8.17.1 -> 8.18.0

### PR DESCRIPTION
**Pull request by vbgl**
## Description of changes

https://gitlab.inria.fr/flocq/flocq/-/blob/flocq-4.1.3/NEWS.md
https://coq.github.io/doc/master/refman/changes.html#changes-in-8-18-0
https://gitlab.inria.fr/fbesson/itauto/-/blob/8.18.0/CHANGELOG

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Test in progress there: https://github.com/coq-community/coq-nix-toolbox/pull/161

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -15720,7 +15720,7 @@ with pkgs;
 
   comby = callPackage ../development/tools/comby { };
 
-  inherit (coqPackages) compcert;
+  inherit (coqPackages_8_17) compcert;
```
Comment (aviallon) [inline pkgs/top-level/all-packages.nix:15723]: Why pinning `compcert`?

**Review by aviallon — APPROVED** [review]
LGTM, a few hashes updates + some dependency updates.
Just have one small remark below.

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (vbgl) [reply pkgs/top-level/all-packages.nix:15723]: It is not available for Coq 8.18 yet.

**Comment by vbgl** [comment]
All checks went through successfully.

File: pkgs/development/coq-modules/itauto/default.nix [hunk 2]
```diff
@@ -33,4 +35,7 @@
 }).overrideAttrs (o: lib.optionalAttrs
   (o.version == "dev" || lib.versionAtLeast o.version "8.16") {
     propagatedBuildInputs = [ coq.ocamlPackages.findlib ];
+} // lib.optionalAttrs
+  (o.version == "dev" || lib.versionAtLeast o.version "8.18") {
+    nativeBuildInputs = with coq.ocamlPackages; [ ocaml findlib dune_3 ];
```
Comment (Zimmi48) [inline pkgs/development/coq-modules/itauto/default.nix:40]: Why not use `useDuneifVersion`?

**Review by Zimmi48 — APPROVED** [review]

**Review by jwiegley — APPROVED** [review]

**Review by wegank — APPROVED** [review]

**Pull request merged**